### PR TITLE
Field migration: host.id -> hw.serial_number

### PR DESCRIPTION
**Pull request by tsellers-r7**
## Description
This PR implements the change discussed in Issue #364. 

The changes:
- changes `host.id` to `hw.serial_number` where the value is a hardware serial number
- changes `host.id` to `host.name` where the value references a host name
- in each case where there is a vendor specific serial number field such as `cisco.serial_number` that value is duplicated in `hw.serial_number` and a test added for it.

## Motivation and Context
The intent is to add clarity to the data that is captured. The field `host.id` was unclear and was being used to capture hostnames as well as serial numbers.

## How Has This Been Tested?
`rspec`, fingerprint example captures.

## Types of changes
- Field rename

## Checklist:

- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

File: identifiers/os_product.txt [hunk 1]
```diff
@@ -116,7 +116,6 @@ IPReach
 IPSO
 IRIX
 Integrated Lights Out Manager
-Integrated Lights Out Manager firmware
```
Comment (tsellers-r7) [inline identifiers/os_product.txt:119]: I rebuilt all of the `identifiers/` files. Apparently this value was no longer used.

File: lib/recog/nizer.rb [hunk 2]
```diff
@@ -264,84 +264,3 @@ def self.best_service_match(matches)
 
 end
 end
-
```
Comment (tsellers-r7) [inline lib/recog/nizer.rb:267]: Out of date comment replaced by the data in the `identifiers/` directory

File: cpe-remap.yaml [hunk 3]
```diff
@@ -309,6 +309,10 @@ mappings:
     hp:
       products:
         ilo: integrated_lights-out
+    kace:
+      vendor: dell
+      products:
+        k1000: kace_k1000_systems_management_appliance
```
Comment (tsellers-r7) [inline cpe-remap.yaml:315]: opportunistic CPE improvement

**Comment by tsellers-r7** [comment]
cc @hdm  @dabdine

**Review by ekelly-rapid7 — APPROVED** [review]

**Review by rkirk-r7 — APPROVED** [review]

**Comment by hdm** [comment]
I agree with changing to `hw.serial_number` as a consistent field, but there may be some adjustment needed on the Rumble side (if not Nexpose too).

**Pull request merged**